### PR TITLE
#37224 endpoint for uploading form files to local file storage

### PR DESCRIPTION
**Pull request by Klaas-Ritense**

### CODE REVIEW

File: process-document/src/main/java/com/ritense/processdocument/service/DocumentDefinitionProcessLinkService.java [hunk 1]
```diff
@@ -21,13 +21,14 @@
 import com.ritense.processdocument.domain.impl.request.DocumentDefinitionProcessLinkResponse;
 import com.ritense.processdocument.domain.impl.request.DocumentDefinitionProcessRequest;
 
+import java.util.List;
 import java.util.Optional;
 
 public interface DocumentDefinitionProcessLinkService {
 
-    DocumentDefinitionProcess getDocumentDefinitionProcess(String documentDefinitionName);
+    List<DocumentDefinitionProcess> getDocumentDefinitionProcess(String documentDefinitionName);
```
Comment (ivo-ritense) [inline process-document/src/main/java/com/ritense/processdocument/service/DocumentDefinitionProcessLinkService.java:29]: This is a breaking change. Is it possible to create this method next to the existing one?

File: process-document/src/main/java/com/ritense/processdocument/service/DocumentDefinitionProcessLinkService.java [hunk 2]
```diff
@@ -21,13 +21,14 @@
...
 
-    Optional<DocumentDefinitionProcessLink> getDocumentDefinitionProcessLink(String documentDefinitionName);
+    Optional<DocumentDefinitionProcessLink> getDocumentDefinitionProcessLink(String documentDefinitionName, String type);
```
Comment (marijnritense) [inline process-document/src/main/java/com/ritense/processdocument/service/DocumentDefinitionProcessLinkService.java:31]: shouldn't this be covered by tests?

File: process-document/src/main/java/com/ritense/processdocument/service/impl/DocumentDefinitionProcessLinkServiceImpl.java [hunk 3]
```diff
@@ -41,24 +43,22 @@ public DocumentDefinitionProcessLinkServiceImpl(DocumentDefinitionProcessLinkRep
...
-    public DocumentDefinitionProcess getDocumentDefinitionProcess(String documentDefinitionName) {
-        var link= documentDefinitionProcessLinkRepository.findByIdDocumentDefinitionName(documentDefinitionName);
+    public List<DocumentDefinitionProcess> getDocumentDefinitionProcess(String documentDefinitionName) {
+        var links = documentDefinitionProcessLinkRepository.findAllByIdDocumentDefinitionName(documentDefinitionName);
 
-        if (link.isPresent()) {
+        return links.stream().map(link -> {
             var processDefinition = repositoryService.createProcessDefinitionQuery()
-                .processDefinitionKey(link.get().getId().getProcessDefinitionKey())
+                .processDefinitionKey(links.get(0).getId().getProcessDefinitionKey())
```
Comment (ivo-ritense) [inline process-document/src/main/java/com/ritense/processdocument/service/impl/DocumentDefinitionProcessLinkServiceImpl.java:51]: Aren't you always still getting the first entry in the list this way? Then what's the point of getting a list?

File: resource/temporary-resource-storage/src/main/kotlin/com/ritense/resource/service/TemporaryResourceStorageService.kt [hunk 4]
```diff
@@ -62,14 +65,31 @@ class TemporaryResourceStorageService {
...
+        }
+        val fileNumber = resourceUuid.leastSignificantBits.toULong()
+        return Path.of(TEMP_DIR.pathString, "temporaryResourceMetadata$fileNumber.json")
+    }
+
+    private fun getResourceIdFromMetaDataFile(tempFile: Path): String {
+        if (!tempFile.nameWithoutExtension.startsWith("temporaryResourceMetadata")) {
+            throw IllegalArgumentException("File is not a temporaryResourceMetadata file: $tempFile")
+        }
+        val fileNumber = tempFile.nameWithoutExtension.substring("temporaryResourceMetadata".length).toULong().toLong()
```
Comment (ivo-ritense) [inline resource/temporary-resource-storage/src/main/kotlin/com/ritense/resource/service/TemporaryResourceStorageService.kt:89]: I'd prefer to have this not be a UUID based on the randomly generated name. The Java documentation says
```
The details as to how the name of the file is constructed is implementation dependent and therefore not specified. Where possible the prefix and suffix are used to construct candidate names in the same manner as the File.createTempFile(String, String, File) method.
```
So i think we can't 100% expect this to be a long. I think we should just return the file name (not the path).

File: process-document/src/main/java/com/ritense/processdocument/service/DocumentDefinitionProcessLinkService.java (hunk 1 above)
Comment (Klaas-Ritense) [reply process-document/src/main/java/com/ritense/processdocument/service/DocumentDefinitionProcessLinkService.java]: Changed

File: process-document/src/main/java/com/ritense/processdocument/service/DocumentDefinitionProcessLinkService.java (hunk 2 above)
Comment (Klaas-Ritense) [reply process-document/src/main/java/com/ritense/processdocument/service/DocumentDefinitionProcessLinkService.java:37]: Added

File: process-document/src/main/java/com/ritense/processdocument/service/impl/DocumentDefinitionProcessLinkServiceImpl.java (hunk 3 above)
Comment (Klaas-Ritense) [reply process-document/src/main/java/com/ritense/processdocument/service/impl/DocumentDefinitionProcessLinkServiceImpl.java]: Changed

File: resource/temporary-resource-storage/src/main/kotlin/com/ritense/resource/service/TemporaryResourceStorageService.kt (hunk 4 above)
Comment (Klaas-Ritense) [reply resource/temporary-resource-storage/src/main/kotlin/com/ritense/resource/service/TemporaryResourceStorageService.kt]: Changed

**Review by ivo-ritense — APPROVED** [review]

**Pull request merged**